sync15-adapter/src/token.rs

```diff
@@ -78,7 +79,8 @@ impl TokenFetcher for TokenServerFetcher {
             debug!("  Response body {}", resp.text().unwrap_or_else(|_| "???".into()));
             // XXX - shouldn't we "chain" these errors - ie, a BackoffError could
             // have a TokenserverHttpError as its cause?
-            if let Some(ms) = resp.headers().get::<RetryAfter>().map(|h| (**h * 1000f64) as u64) {
+            // XXX - we are silentely dropping UTF8 parsing/number parsing errors here.
+            if let Some(ms) = resp.headers().get(RETRY_AFTER).and_then(|v| v.to_str().ok()).and_then(|s| s.parse::<f64>().ok()).map(|f| (f * 1000f64) as u64) {
```

Comment:
We should see if we can enter this block if there's a retry after header at all. Maybe make up a backoff time if we fail to get a reasonable one? Doesn't seem good to silently ignore it.